lib/hashes.md

```diff
@@ -5,15 +5,16 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is a Hash?
 
-Replace me with your response...
+Hashes are sort of like dictionaries. It is a collection of keys and values where you can refer to a value using the associated key, or vice versa.
```

Comment:
What I meant by that is that you can search through all elements in a Hash to find the value for a given key, or to find all the keys for a given value. However, I am discovering that it does not work as well to search by value as it does to search by key (which I believe might be your point...).

As an example, let's say I had a list of students (Frank, Joe, Anna, and Lacey). I create a Hash where their names are my keys and their ages are my values.

If I wanted to find out Frank's age, that would be easy using a Hash. I could retrieve the value for the key "Frank".

But what if I wanted to find the name of the student who was 25 years old? I could use this code to retrieve the key given the value of 25.

That code would look like this: 

test_hash = {"Frank" => 25, "Joe" => 30, "Anna" => 78, "Lacey" => 47}
    puts test_hash.key(25)

I tried this on repl.it and it returned "Frank" when ran.

However, what I am discovering is that it is less simple to retrieve ALL the keys in a Hash that match a given value. So, let's say my class list and corresponding Hash looked like this:

test_hash = {"Frank" => 25, "Joe" => 30, "Anna" => 78, "Lacey" => 47, "Ted" => 25}

And I wanted to retrieve all keys in the Hash that had a value of 25 (or find all students in the class who are 25 years old).

This does not seem to be something as easy to do with Hashes as I originally thought...

I have found options to invert the keys to values, or to use .map, but neither of those seemed to be a simple way to perform that command...

I don't encounter this issue retrieving the value for a given key because there cannot be duplicate keys. 

Am I simply unaware of an iterator command that would help me do this, (I am still trying to understand iterators a bit) or am I correct that it's not ideal to use a Hash this way?